knapsack: split introduction and instructions (#2440)
exercism/problem-specifications: exercises/knapsack/introduction.md

# Introduction

Bob is a thief.
After months of careful planning, he finally manages to crack the security systems of a fancy store.

In front of him are many items, each with a value and weight.

> Whilst Bob would gladly take all of the items, his knapsack can only hold so much weight.

Reviewer: Whilst is not a word I see used often. What about the simpler,

```suggestion
Bob would gladly take all of the items, but his knapsack can only hold so much weight.
```

> Therefore, Bob has to carefully consider which items to take to maximize their value.

Reviewer 2: Slightly confusing: the items each have a fixed value that cannot be 'maximized'.

```suggestion
Therefore, Bob has to carefully consider which items to take to maximize his profit.
```

(The word _profit_ sounds slightly unnatural to me here; native speakers please check this.)

Reviewer 3: I don't believe the instructions indicate we can't change the value of a single item. Perhaps, we should write "maximize their collective value."

Profit is awkward here because that'd be in the future when Bob successfully sells the items he stole. We're still at the point where he's stealing the items so we can't assess profit. We can assess value though at any time.

Reviewer 4: > Bob has to carefully consider which items to take so that the total value of his selection is maximized.

Author: ```suggestion
Bob has to carefully consider which items to take so that the total value of his selection is maximized.
```
